File: iPad/VerBank-CSTSv3-IOSClientApi/ClientStation/api/doc/operator/system/SystemDocOperator.m
//
//  SystemDocOperator.m
//  VerBank-IOSClientStation
//
//  Created by Allone on 15/8/31.
//  Copyright (c) 2015年 zhanglei. All rights reserved.
//

#import "SystemDocOperator.h"

#import "API_IDEventCaptain.h"

#import "CSTSCaptain.h"
#import "APIDoc.h"

#import "IP_TRADESERV5010.h"
#import "OP_TRADESERV5010.h"
#import "IP_TRADESERV5011.h"
#import "OP_TRADESERV5011.h"

static SystemDocOperator *instance = nil;

@implementation SystemDocOperator

+ (id)getInstance {
    if (instance == nil) {
        instance = [[SystemDocOperator alloc] init];
    }
    return instance;
}

- (Boolean)loadBasicCurrency {
    //    API_IDEventCaptain fireEventChanged:<#(NSString *)#> eventData:<#(id)#>]
    IP_TRADESERV5010 *ip =[[IP_TRADESERV5010 alloc] init];
    OPFather *opFather = [[CSTSCaptain getInstance] trade:ip];
    
    if (![opFather isSuccessed]) {
        return false;
    }
    
    OP_TRADESERV5010 *op = (OP_TRADESERV5010 *)opFather;
    [[APIDoc getSystemDocCaptain] resetBasicCurrencys:[op getBasicCurrencyArray]];
    return true;
}

- (Boolean)loadSystemConfig {
    IP_TRADESERV5011 *ip = [[IP_TRADESERV5011 alloc] init];
    OPFather *opFather = [[CSTSCaptain getInstance] trade:ip];
    
    if (![opFather isSuccessed]) {
        return false;
    }
    
    OP_TRADESERV5011 *op = (OP_TRADESERV5011 *)opFather;
    [[APIDoc getSystemDocCaptain] setSystemConfig:[op getSystemConfig]];
    return true;
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/doc/operator/system/SystemDocOperator.java
package jedi.ex01.client.station.api.doc.operator.system;

import jedi.ex01.CSTS3.comm.ipop.IP_TRADESERV5010;
import jedi.ex01.CSTS3.comm.ipop.IP_TRADESERV5011;
import jedi.ex01.CSTS3.comm.ipop.IP_TRADESERV5016;
import jedi.ex01.CSTS3.comm.ipop.OPFather;
import jedi.ex01.CSTS3.comm.ipop.OP_TRADESERV5010;
import jedi.ex01.CSTS3.comm.ipop.OP_TRADESERV5011;
import jedi.ex01.CSTS3.comm.ipop.OP_TRADESERV5016;
import jedi.ex01.client.station.api.CSTS.CSTSCaptain;
import jedi.ex01.client.station.api.doc.DataDoc;
import jedi.ex01.client.station.api.event.API_IDEventCaptain;
import jedi.ex01.client.station.api.event.API_IDEvent_NameInterface;

public class SystemDocOperator {
	private static SystemDocOperator systemDocOperator=new SystemDocOperator();
	private SystemDocOperator(){
	}
	
	public boolean loadBasicCurrency(){
		IP_TRADESERV5010 ip = new IP_TRADESERV5010();
		OPFather opfather = CSTSCaptain.getInstance().trade(ip);
		if (!opfather.isSucceed()) {
			return false;
		}
		OP_TRADESERV5010 op = (OP_TRADESERV5010) opfather;
		DataDoc.getInstance().resetBasicCurrencys(op.getBasicCurrencyVec());
		return true;
	}
	
	public boolean loadSystemConfig(){
		IP_TRADESERV5011 ip = new IP_TRADESERV5011();
		OPFather opfather = CSTSCaptain.getInstance().trade(ip);
		if (!opfather.isSucceed()) {
			return false;
		}
		OP_TRADESERV5011 op = (OP_TRADESERV5011) opfather;
		DataDoc.getInstance().setSystemConfig(op.getSystemConfig());
		return true;
	}
	
//	public boolean loadTradeTypeConfigs(){
//		IP_TRADESERV5016 ip = new IP_TRADESERV5016();
//		OPFather opfather = CSTSCaptain.getInstance().trade(ip);
//		if (!opfather.isSucceed()) {
//			return false;
//		}
//		OP_TRADESERV5016 op = (OP_TRADESERV5016) opfather;
//		DataDoc.getInstance().setCfdTradeTypeConfig(op.getTradeTypeConfigVec()[0]);
//		API_IDEventCaptain.fireEventChanged(API_IDEvent_NameInterface.DATA_ON_TradeType_Changed);
//		return true;
//	}
	
	public static SystemDocOperator getSystemDocOperator() {
		return systemDocOperator;
	}
}
